### PR TITLE
vector : Add Laplacian function

### PR DESCRIPTION
**Pull request by ArifAhmed1995**
This PR adds the laplacian function for both vector and scalar fields in the vector module.

https://en.wikipedia.org/wiki/Laplace_operator
https://en.wikipedia.org/wiki/Vector_Laplacian

Example :
``` 
>>> from sympy.vector import laplacian
>>> from sympy.vector.coordsysrect import CoordSysCartesian
>>> C = CoordSysCartesian('C')
>>> i, j, k = C.base_vectors()
>>> x, y, z = C.base_scalars()
>>> laplacian(x*y**2*z, C)
2*C.x*C.z
>>> laplacian(x*y**2*z*(i + j + k), C)
2*C.x*C.z*C.i + 2*C.x*C.z*C.j + 2*C.x*C.z*C.k
```

### CODE REVIEW

**Comment by ArifAhmed1995** [comment]
@moorepants Can you review this PR ?

**Comment by asmeurer** [comment]
Seems it would be better for `laplacian` to just detect if the input is a vector or scalar and apply the appropriate computation.

**Comment by ArifAhmed1995** [comment]
@asmeurer Done.

**Comment by ArifAhmed1995** [comment]
@asmeurer Merge this PR now ?

**Comment by nicoguaro** [comment]
I am interested in this PR, can we add other operators like curl, gradient of a vector field or divergence of a tensor field?

**Comment by gxyd** [comment]
@szymag can you please review this PR?

**Comment by szymag** [comment]
The code looks good to me but I suggest to add tests for different types of coordinate system.

The output could be easily checked with wikipedia:
https://en.wikipedia.org/wiki/Laplace_operator#Coordinate_expressions

File: sympy/vector/functions.py [hunk 1]
```diff
@@ -173,6 +174,36 @@ def directional_derivative(field, direction_vector):
         return S(0)
 
 
+def laplacian(expr):
+    """
+    Returns the laplacian of the given field computed wrt
```
Comment (szymag) [inline sympy/vector/functions.py]: *wrt* means `with respect to`, right?
For base scalars it sounds strange to me.
I suggest:
*Returns the laplacian of the given field computed **in terms of** the base scalars of the given coordinate system*

Comment (gxyd) [reply sympy/vector/functions.py]: Yes, 'wrt' means 'with respect to' here. I think you are right, *in terms of* is better suited here.

How about we instead have *Return the sum of second order partial derivatives of 'f' with respect to the base scalars* ?

Comment (szymag) [reply sympy/vector/functions.py]: It's good proposition but I'd prefer something which use concept from vector analysis.

**Comment by nicoguaro** [comment]
Wouldn't it fail for systems that are not orthogonal?

**Comment by szymag** [comment]
In `sympy.vector` we are dealing with only orthogonal system. for example `spherical` system is orthogonal one. 
I don't know what your suggestion exactly mean

**Comment by nicoguaro** [comment]
Well, I thing that the main problem would be the definition of the divergence as as dot product. Dot product is computed straightforward in Cartesian Coordinates, but not in general. Besides that, I don't see why somebody can't define a basis that is not normalized or orthogonal.

**Comment by gxyd** [comment]
> Besides that, I don't see why somebody can't define a basis that is not normalized or orthogonal.

Would that require more enhancement in the current vector module of SymPy? Because if it does then you can open up an issue about it.

**Comment by nicoguaro** [comment]
It probably require changes in the module. I am not that sure. The way I think of those operators are in curvilinear coordinates and scale factors. In the following notebok

http://nbviewer.jupyter.org/github/nicoguaro/notebooks_examples/blob/master/Vectors_operators.ipynb

I use that approach for different operators.

I should also add, that I am interested in using this operators in tensors that might be of higher order, mainly, second order tensors and vectors, though.

**Comment by gxyd** [comment]
This has passed the test. I am merging this as is.

@nicoguaro you can open an issue, explaining what you did here. We can discuss further on the issue.

**Pull request merged**